go.mod

```diff
@@ -1,7 +1,7 @@
 module github.com/open-policy-agent/opa-envoy-plugin
 
-go 1.21
-toolchain go1.22.5
```

Comment:
Thanks for updating the Go version. The min Go version to build this module has changed. Why is that needed?

Reply:
@ashutosh-narkar - As mentioned in the PR's description, this to ensure that it aligns with version using by main OPA binary (Ref: https://github.com/open-policy-agent/opa/blob/main/.go-version)

And also to mitigate the CVEs listed in the PR.

Reply:
I get that we should update the Go version. I'm asking about the minimum version to build the module has also updated and wanted to know why that is? If you check OPA's [go.mod](https://github.com/open-policy-agent/opa/blob/main/go.mod#L3) file the minimum version was not updated.

Reply:
@ashutosh-narkar  - removed the update of minimum go version.